frontend/src/contexts/Session.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+  This file is part of Edgehog.
+
+  Copyright 2025 SECO Mind Srl
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  SPDX-License-Identifier: Apache-2.0
+*/
+
+import React, { createContext, useCallback, useContext, useState } from "react";
+import Cookies from "js-cookie";
+import _ from "lodash";
+
+type Session = {
+  tenantSlug: string;
+  authToken: string | null;
+};
+
+const SESSION_CONFIG_VERSION = 1;
+
+type SessionContextValue = {
+  session: Session | null;
+  saveSession: (session: Session | null, persist?: boolean) => void;
+};
```

Comment:
In my opinion, components using this context shouldn't care if the session is stored or not.
Also, a session should always be present. If there is no previous session, a new one will be created right away starting from scratch.

```suggestion
type SessionContextValue = {
  session: Session;
  updateSession: (session: Session) => void;
};
```